python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -231,6 +249,79 @@ def run(self, splash=None, version=None):
         result = app.exec_()
         return result
 
+    def uses_legacy_authentication(self):
+        """
+        Returns if the Desktop's bootrap code uses the tk-framework-login for
+        logging in.
+
+        :returns: True the bootstrap logic is older than 1.1.0, False otherwise.
+        """
+        return StrictVersion(self.app_version) < StrictVersion("1.1.0")
+
+    def create_legacy_login_instance(self):
+        """
+        Creates the pre-bootstrap-1.1.0 ShotgunLogin used to authenticate.
+
+        :raises ImportError: Thrown if the module is not available.
+
+        :returns: A tk-framework-login's ShotgunLogin instance.
+        """
+        try:
+            from python import ShotgunLogin
+        except ImportError:
+            self.log_exception("Could not import tk-framework-login")
+            raise
+        else:
+            return ShotgunLogin.get_instance_for_namespace("tk-desktop")
+
+    def _assert_authentication_fault_exists(self, splash):
+        """
+        Asserts that we importing the right version of shotgun_api3. If we are
+        not, the user will be shown a countdown and the app will be restarted.
+        """
+        from tank_vendor import shotgun_api3
```

Comment:
The else branch is not possible. The tk-desktop 2.0.0 engine will download if and only if you have Core 0.16.0 and higher. So basically, the test is good enough I think and less complicated than finding an info.yml file, parsing it and comparing values.

If the Internet is down, then you won't download the core 0.16 and you won't be running the tk-desktop 2.0